fv3gfs/util/communicator.py

```diff
@@ -468,64 +465,31 @@ def start_halo_update(self, quantity: Quantity, n_points: int) -> HaloUpdateRequ
         Returns:
             request: an asynchronous request object with a .wait() method
         """
-        if n_points == 0:
-            raise ValueError("cannot perform a halo update on zero halo points")
-        CubedSphereCommunicator._device_synchronize()
-        tag = self._get_halo_tag()
-        recv_data = self._Irecv_halos(quantity, n_points, tag=tag)
-        send_data = self._Isend_halos(quantity, n_points, tag=tag)
-        return HaloUpdateRequest(send_data, recv_data, self.timer)
-
-    def _Isend_halos(
-        self, quantity: Quantity, n_points: int, tag: int = 0
-    ) -> _HaloRequestSendList:
-        send_data = []
-        for boundary in self.boundaries.values():
-            with self.timer.clock("pack"):
-                source_view = boundary.send_view(quantity, n_points=n_points)
-                # sending data across the boundary will rotate the data
-                # n_clockwise_rotations times, due to the difference in axis orientation.\
-                # Thus we rotate that number of times counterclockwise before sending,
-                # to get the right final orientation
-                source_view = rotate_scalar_data(
-                    source_view,
-                    quantity.dims,
-                    quantity.np,
-                    -boundary.n_clockwise_rotations,
-                )
-            send_data.append(
-                self._Isend(
-                    self._maybe_force_cpu(quantity.np),
-                    source_view,
-                    dest=boundary.to_rank,
-                    tag=tag,
-                )
+        if isinstance(quantity, Quantity):
+            quantities = [quantity]
+        else:
+            quantities = quantity
+
+        specifications = []
+        for quantity in quantities:
+            specification = HaloUpdateSpec(
```

Comment:
Well since we are looking at creating halo spec without the need for quantity via a future gt4py interface to reach strides information, I am not sure it make sense to have it as a member fn.